ocular/src/keyring.rs

```diff
@@ -0,0 +1,752 @@
+use bech32::{Bech32, ToBase32};
+use bip32::PrivateKey;
+use bip39::{Mnemonic, Seed};
+use ripemd::Ripemd160 as Ripemd;
+use serde::{Deserialize, Serialize};
+use sha2::Digest as Sha2Digest;
+use sha2::Sha256;
+use signatory::{
+    pkcs8::der::Document, pkcs8::EncodePrivateKey, pkcs8::LineEnding, pkcs8::PrivateKeyDocument,
+    FsKeyStore, KeyName,
+};
+use std::{env, fs, path::Path};
+
+use crate::error::KeyStoreError;
+
+// Constants
+// TODO: Move to independant constants file if reused elsewhere
+static COSMOS_BASE_DERIVATION_PATH: &str = "m/44'/118'/0'/0/0";
+static COSMOS_ADDRESS_PREFIX: &str = "cosmos";
```

Comment:
These are constants, so use `const`:

```suggestion
const COSMOS_BASE_DERIVATION_PATH: &str = "m/44'/118'/0'/0/0";
const COSMOS_ADDRESS_PREFIX: &str = "cosmos";
```

Reply:
Totally missed that on earlier passes, updated.